src/widgets/editor.rs

```diff
@@ -118,9 +120,77 @@ impl Future for Editor {
 
 impl Editor {
     /// Handle keyboard and mouse events
-    pub fn handle_input(&mut self, event: TermionEvent) {
-        if let Some(view) = self.views.get_mut(&self.current_view) {
-            view.handle_input(event)
+    pub fn handle_input(&mut self, event: Event) {
+        // We have to remove and insert again, to beat the borrow-checker
```

Comment:
I'm confused be this comment which does not seem to match what the code does, unless I'm missing something?

Reply:
ah yes thats a left-over comment. Should be removed!